File: backend/src/main/java/com/happy/friendogly/chat/dto/response/ChatMessageResponse.java
package com.happy.friendogly.chat.dto.response;

import com.happy.friendogly.chat.domain.MessageType;
import com.happy.friendogly.member.domain.Member;
import java.time.LocalDateTime;

public record ChatMessageResponse(
        MessageType messageType,
        Long senderMemberId,
        String senderName,
        String content,
        LocalDateTime createdAt,
        String profilePictureUrl
) {

    public ChatMessageResponse(MessageType messageType, String content, Member senderMember, LocalDateTime createdAt) {
        this(
                messageType,
                senderMember.getId(),
                senderMember.getName().getValue(),
                content,
                createdAt,
                senderMember.getImageUrl()
        );
    }
}

File: android/app/src/main/java/com/happy/friendogly/remote/model/response/ChatMessageResponse.kt
package com.happy.friendogly.remote.model.response

import com.happy.friendogly.remote.util.JavaLocalDateTimeSerializer
import kotlinx.serialization.Serializable
import java.time.LocalDateTime

@Serializable
data class ChatMessageResponse(
    val messageType: MessageTypeResponse,
    val senderMemberId: Long,
    val senderName: String,
    val content: String?,
    @Serializable(with = JavaLocalDateTimeSerializer::class)
    val createdAt: LocalDateTime,
    val profilePictureUrl: String?,
)
